### PR TITLE
chore(web): hide and show layer

### PR DESCRIPTION
**Pull request by mkumbobeaty**
# Overview

## What I've done
- implement hide/show layer UI
- add hide/show functionalities

### CODE REVIEW

File: web/src/beta/lib/core/mantle/types/index.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export type LayerSimple = {
   properties?: any;
   defines?: Record<string, string>;
   events?: Events;
+  visible?: boolean;
```
Comment (KaWaite) [inline web/src/beta/lib/core/mantle/types/index.ts:29]: `visible` is included in `LayerCommon` 👍🏻

File: web/src/beta/lib/core/Map/ClusteredLayers/index.tsx [hunk 2]
```diff
@@ -65,7 +65,7 @@ export default function ClusteredLayers({
   const renderLayer = useCallback(
     (layer: Layer) => {
       const a = atomMap?.get(layer.id);
-      return !layer.id || !a ? null : (
+      return !layer.id || !a || !layer.visible ? null : (
```
Comment (KaWaite) [inline web/src/beta/lib/core/Map/ClusteredLayers/index.tsx:68]: @keiya01 Should this be like this? Or should checking visible, and including/removing happen at the atomMap level?

File: web/src/beta/features/Editor/useLayers.ts [hunk 3]
```diff
@@ -18,7 +18,8 @@ export type LayerAddProps = {
 
 export type LayerNameUpdateProps = {
   layerId: string;
-  name: string;
+  name?: string;
+  visible?: boolean;
```
Comment (KaWaite) [inline web/src/beta/features/Editor/useLayers.ts:22]: This type is `LayerNameUpdateProps`. If you want to make this a general layer update prop and callback please update the naming 🙏🏻

File: web/src/beta/components/ListItem/index.tsx [hunk 4]
```diff
@@ -42,6 +46,10 @@ const ListItem: FC<Props> = ({
           children
         )}
       </Inner>
+
+      <HideLayer onClick={onLayerUpdate} isSelected={isSelected}>
+        {value}
+      </HideLayer>
```
Comment (KaWaite) [inline web/src/beta/components/ListItem/index.tsx:52]: Is there any problem with this being passed inside the children node? That way you don't have to change this ListItem at all..

**Review by pyshx — COMMENTED** [review]
@KaWaite @keiya01 Do we want the visibility to be handled on frontend level? We can also do it by making mutation to layer. Not sure if we want to retain visibility for layers when a session is finished 🤔

File: web/src/beta/lib/core/Map/ClusteredLayers/index.tsx (hunk 2 above)
Comment (keiya01) [reply web/src/beta/lib/core/Map/ClusteredLayers/index.tsx:68]: This should be handled in [here](https://github.com/reearth/reearth/blob/main/web/src/beta/lib/core/engines/Cesium/Feature/index.tsx#L146) or [here](https://github.com/reearth/reearth/blob/main/web/src/beta/lib/core/engines/Cesium/Feature/index.tsx#L172).

Comment (mkumbobeaty) [reply web/src/beta/lib/core/Map/ClusteredLayers/index.tsx:68]: @keiya01 if it in [here](https://github.com/reearth/reearth/blob/main/web/src/beta/lib/core/engines/Cesium/Feature/index.tsx#L172) I will need to re-render the component to update the map

Comment (mkumbobeaty) [reply web/src/beta/lib/core/Map/ClusteredLayers/index.tsx:68]: @keiya01 I try to put it here but the layer is not removed in map until I restart the page

File: web/src/beta/lib/core/engines/Cesium/Feature/index.tsx [hunk 5]
```diff
@@ -99,12 +100,13 @@ export default function Feature({
       !!props.sceneProperty?.light?.sphericalHarmonicCoefficients;
     const useSceneSpecularEnvironmentMaps = !!props.sceneProperty?.light?.specularEnvironmentMaps;
 
+    const isVisible = layer.layer.visible !== false || isHidden;
```
Comment (keiya01) [inline web/src/beta/lib/core/engines/Cesium/Feature/index.tsx:103]:
```suggestion
    const isVisible = layer.layer.visible !== false && !isHidden;
```

**Review by keiya01 — APPROVED** [review]

File: web/src/beta/components/ListItem/index.tsx [hunk 6]
```diff
@@ -66,11 +67,16 @@ const Wrapper = styled.div`
...
+const Inner = styled.button<{
+  border?: boolean;
+  isSelected?: boolean;
+  clamp?: Clamp;
+}>`
   display: flex;
   width: 100%;
   min-height: 38px;
   align-items: center;
+  color: #e0e0e0;
```
Comment (KaWaite) [inline web/src/beta/components/ListItem/index.tsx:79]: Can you use theme please 🙏🏻

**Review by KaWaite — APPROVED** [review]

**Review by pyshx — APPROVED** [review]

**Review by pyshx — APPROVED** [review]

**Review by KaWaite — APPROVED** [review]

**Pull request merged**